lib/stateful/animation.dart

```diff
@@ -0,0 +1,288 @@
+import 'dart:async';
+import 'dart:math' as math;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+
+class StatefulAnimation extends StatefulWidget {
+  StatefulAnimation({
+    Key key,
+  }): super(key: key);
+
+  @override
+  _StatefulAnimationState createState() => _StatefulAnimationState();
+}
+
+class _StatefulAnimationState extends State<StatefulAnimation> {
+  ValueNotifier<Color> _colorListenable;
```

Comment:
Why is this inside StatefulAnimation rather than Example?

Reply:
It's to show that the ValueListenable is coming via the widget rather than being local to the widget (in which case you wouldn't use a ValueListenable in the first place). Ideally it would be in the root widget in main.dart but I couldn't figure out how to make the hooks widget do it.

Reply:
I'll merge it in as is, I can place the ValueNotifier higher in the tree to make more clear the purpose later today.

Reply:
Might be fine to have them in the example as two different widgets just to show how it would be done in each mode. That keeps the weirdness out of the root.

Reply:
I'm still a bit weirded out by the presence of the `ValueListenable<Color> color`. 
It feels like some unrelated architecture decisions are leaking in the example.